airsonic-main/src/main/java/org/airsonic/player/service/UPnPService.java

```diff
@@ -61,10 +62,17 @@
 
     @Autowired
     private SettingsService settingsService;
+    
     private UpnpService upnpService;
+    
     @Autowired
     @Qualifier("dispatchingContentDirectory")
     private CustomContentDirectory dispatchingContentDirectory;
+    
+    @Autowired
+    @Value("#{ systemProperties['UPNP_PORT'] ?: T(org.airsonic.player.service.SettingsService).DEFAULT_UPNP_PORT }")
+    private int port;
```

Comment:
This doesn't match the existing methodology for managing settings. Should instead use a getter through settingservice.

Reply:
`T(org.airsonic.player.service.SettingsService).DEFAULT_UPNP_PORT`

is the public variable in the SettingsService. do you want it to be a getter? There's no data encapsulation happening

Reply:
Yes we should be consistent with all settings. In addition, we should be fetching from the settings service for the value, not from the environment.